pkg/ibmcsidriver/ibm_csi_driver.go

```diff
@@ -108,12 +110,30 @@ func (icDriver *IBMCSIDriver) SetupIBMCSIDriver(provider cloudProvider.CloudProv
 	icDriver.cs = NewControllerServer(icDriver, provider)
 
 	icDriver.logger.Info("Successfully setup IBM CSI driver")
+
 	// Set up Region
 	regionMetadata, err := nodeInfo.NewNodeMetadata(lgr)
 	if err != nil {
 		return fmt.Errorf("Controller_Helper: Failed to initialize node metadata: error: %v", err)
 	}
 	icDriver.region = regionMetadata.GetRegion()
+
+	// get the session
+	icDriver.rfsEnabled = false
+	session, err := provider.GetProviderSession(context.Background(), lgr)
+	if err != nil {
+		icDriver.logger.Warn("Cannot fetch session for verifying RFS profile")
+		return nil
+	}
+
+	err = session.GetVolumeProfileByName(RFSProfile)
+	if err != nil {
+		icDriver.rfsEnabled = false
```

Comment:
Its not required to set false again
also warning msg need to improve like user don't have access to RFS profile so RFS share creation will not work, user need to be allowed for RFS profile from VPC IaaS team etc

Reply:
This error message is not visible to user but the one we formatted is in the controller createVolume. Still I have changed this logic